repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/identityv3/authz/IdentityV3AuthZBypassTest.groovy

```diff
@@ -0,0 +1,113 @@
+                'identity service', null, fakeIdentityV3Service.handler)
+        //targetPort = properties.targetPort
+    }
+
+    def cleanupSpec() {
+        if (deproxy)
+            deproxy.shutdown()
+
+        if (repose)
+            repose.stop()
+    }
+
+    @Unroll("With Project ID: #requestProject, return from identity with response project: #responseProject, and role: #serviceAdminRole, return 200")
+    def "when authenticating project id and roles that bypass"() {
+        given:
+        fakeIdentityV3Service.with {
+            client_token = UUID.randomUUID().toString()
+            tokenExpiresAt = DateTime.now().plusDays(1)
+            client_projectid = responseProject
+            service_admin_role = serviceAdminRole
+            client_userid = requestProject
+            endpointUrl = "localhost"
+            servicePort = targetPort
+        }
+
+        when:
+        "User passes a request through repose with request tenant: $requestProject, response tenant: $responseProject in a bypassed role = $serviceAdminRole"
+        MessageChain mc = deproxy.makeRequest(
+                url: "$reposeEndpoint/servers/$requestProject/",
+                method: 'GET',
+                headers: [
+                        'content-type'   : 'application/json',
+                        'X-Subject-Token': fakeIdentityV3Service.client_token
+                ]
+        )
+
+        then: "Request body sent from repose to the origin service should contain"
+        mc.receivedResponse.code == responseCode
+        mc.handlings.size() == 1
+        mc.handlings[0].endpoint == originEndpoint
+        def request2 = mc.handlings[0].request
+        request2.headers.getFirstValue("x-forwarded-for") == "127.0.0.1"
+        //if projectId as tenantid in keystonev2 we expect should behave the same
+        //then x-project-id should contain default project id - this isn't sure an issue
+        //Comment this out for now will state in issue REP-3006
+        //TODO - enable this after issue fix
+        //request2.headers.getFirstValue("x-project-id") == responseProject.toString()
+        request2.headers.contains("x-token-expires")
+        request2.headers.getFirstValue("x-pp-user") == fakeIdentityV3Service.client_username + ";q=1.0"
+        request2.headers.contains("x-roles")
+        request2.headers.getFirstValue("x-authorization") == "Proxy"
+        request2.headers.getFirstValue("x-user-name") == "username"
+
+        mc.receivedResponse.headers.contains("www-authenticate") == false
+
+        where:
+        requestProject | responseProject | serviceAdminRole      | endpoint        | responseCode
+        717            | 717             | "not-admin"           | "localhost"     | "200"
+        718            | 719             | "service:admin-role1" | "rackspace.com" | "200"
+        719            | 720             | "service:admin-role2" | "rackspace.com" | "200"
```

Comment:
I was going to say we could add more test cases, but thinking about it, those are probably covered in other IdentityV3 tests. These just cover the new case -- bypassing endpoint authorization for service admins.